### DISCUSSION TITLE
Rabbitmq 3.12 mqtt connection test,but now can only connect to about 65519,how can i connection more?

### DISCUSSION THREAD

**Discussion by springmvcGitHub** [rabbitmq/rabbitmq-server#8494] [Other]
Hello, now I am doing mqtt connection test, hoping to reach 50w connection number, but now can only connect to about 65519.
my test scenario is as follows:
### server:
rabbitmq one node,hardware configuration 32core64G，ip address:192.168.100.10
fs.file-max = 3187211
fs.nr_open = 1048576
ulimit -n = 1048576
net.core.somaxconn = 9000
net.ipv4.tcp_max_syn_backlog = 8096
net.core.netdev_max_backlog = 2000
net.ipv4.tcp_fin_timeout = 60
net.nf_conntrack_max = 524288
net.netfilter.nf_conntrack_max = 524288
net.netfilter.nf_conntrack_tcp_timeout_time_wait = 120
net.ipv4.ip_local_port_range = 32768	60999
net.core.rmem_default = 212992
net.core.wmem_default = 212992
net.core.rmem_max = 212992
net.core.wmem_max = 212992
net.core.optmem_max = 20480
net.ipv4.tcp_rmem = 4096	87380	6291456
net.ipv4.tcp_wmem = 4096	16384	4194304

### client
client has 6 services,One pod per service,4core8G，ipaddress:192.168.100.1-192.168.100.6

**Comment by lukebakken** [comment]
This isn't enough information for us to assist you. You do not tell us what errors you see, or provide any logs.

You should expand the server's local port range, to start with, and should do this on all your client machines as well.

```
net.ipv4.ip_local_port_range = 1024 65535
```

TCP connections are 4-tuples of source IP, source port, destination IP and destination port. If you expect to reach 50 million connections (I don't know what "50w" means) you'll have to ensure that you have enough unique combinations of source IP, source port, destination IP and destination port to reach that number.
